js/ui/toolbar/ui.toolbar.utils.js

```diff
@@ -0,0 +1,43 @@
+import $ from '../../core/renderer';
+
+const BUTTON_GROUP_CLASS = 'dx-buttongroup';
+const TOOLBAR_ITEMS = ['dxAutocomplete', 'dxButton', 'dxCheckBox', 'dxDateBox', 'dxMenu', 'dxSelectBox', 'dxTabs', 'dxTextBox', 'dxButtonGroup', 'dxDropDownButton'];
+
+const getItemInstance = function($element) {
+    const itemData = $element.data && $element.data();
+    const dxComponents = itemData && itemData.dxComponents;
+    const widgetName = dxComponents && dxComponents[0];
+
+    return widgetName && itemData[widgetName];
+};
+
+export function toggleItemFocusableElementTabIndex(context, item) {
+    if(!context) return;
+
+    const $item = context._findItemElementByItem(item);
+    if(!$item.length) {
+        return;
+    }
+
+    const itemData = context._getItemData($item);
+    const isItemNotFocusable = !!(itemData.options?.disabled || itemData.disabled || context.option('disabled'));
+
+    const { widget } = itemData;
+
+    if(widget && TOOLBAR_ITEMS.indexOf(widget) !== -1) {
+        const $widget = $item.find(widget.toLowerCase().replace('dx', '.dx-'));
+        if($widget.length) {
+            const itemInstance = getItemInstance($widget);
+            const $focusTarget = widget === 'dxDropDownButton'
+                ? itemInstance._focusTarget().find(`.${BUTTON_GROUP_CLASS}`)
+                : itemInstance?._focusTarget?.() || $(itemInstance.element());
+
+            const tabIndex = itemData.options?.tabIndex;
+            if(isItemNotFocusable) {
+                $focusTarget.attr('tabIndex', -1);
+            } else {
+                $focusTarget.attr('tabIndex', tabIndex ? tabIndex : 0);
```

Comment:
Default value is similar to this value?
https://github.com/DevExpress/DevExtreme/blob/3d5fff697784a0eb362d6dd63bfe42177a4528cc/js/ui/widget/ui.widget.js#L53